src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -252,4 +821,232 @@ public Request parse(String commandLine) {
+           if (uriProxy.getPort() != -1) {
+               request.setProxyServer("port", String.valueOf(uriProxy.getPort()));
+           } else {
+               request.setProxyServer("port", "1080");
+           }
+       } catch (URISyntaxException e) {
+           LOGGER.error("string '{}' cannot be converted to a URL", proxyServerParameters);
+           throw new IllegalArgumentException(proxyServerParameters + " cannot be converted to a URL");
+       }
+   }
+
+   /**
+    * Set the username and password of proxy server
+    *
+    * @param request               http request
+    * @param proxyServerUserPasswd the username and password of proxy server
+    */
+   private void setProxyServerUserInfo(Request request, String proxyServerUserPasswd) {
+       if (proxyServerUserPasswd.contains(":")) {
+           String[] userInfo = proxyServerUserPasswd.split(":");
+           request.setProxyServer("username", userInfo[0]);
+           request.setProxyServer("password", userInfo[1]);
+       }
+   }
+
+   /**
+    * Get post data by different type of data option
+    *
+    * @param postdata       the post data
+    * @param dataOptionName the different option of "--data"
+    * @return the post data
+    */
+   private String getPostDataByDifferentOption(String postdata, String dataOptionName) {
+       if (dataOptionName.equals("data-urlencode")) {
+           postdata = encodePostdata(postdata);
+       } else {
+           if (postdata.contains("@") && !dataOptionName.equals("data-raw")) {
+               postdata = postdata.replace("@", "");
+               postdata = readFromFile(postdata);
+               if (!dataOptionName.equals("data-binary")) {
+                   postdata = deleteLineBreak(postdata);
+               }
+           }
+       }
+       return postdata;
+   }
+
+   /**
+    * Encode the post data
+    *
+    * @param postdata the post data
+    * @return the result of encoding
+    *
+    */
+   private String encodePostdata(String postdata) {
+       String res = null;
+       if (postdata.contains("@")) {
+           String[] arr = postdata.split("@");
+           try {
+               res = URLEncoder.encode(readFromFile(arr[1]), StandardCharsets.UTF_8.name());
+           } catch (UnsupportedEncodingException e) {
+               LOGGER.error("string '{}' cannot be encoded", readFromFile(arr[1]));// NOSONAR
+           }
+           if (!arr[0].isEmpty()) {
+               res = arr[0] + "=" + res;
+           }
+       } else {
+           if (!postdata.contains("=")) {
+               try {
+                   res = URLEncoder.encode(postdata, StandardCharsets.UTF_8.name());
+               } catch (UnsupportedEncodingException e) {
+                   LOGGER.error("string '{}' cannot be encoded", postdata);
+                   throw new IllegalArgumentException(postdata + " cannot be encoded");
+               }
+           } else {
+               StringBuilder urlAfterEncoding = new StringBuilder();
```

Comment:
Instead of using a StringBuilder we could directly return the concatenated string parts (and get rid of the `res` variable, if we return the results early above).